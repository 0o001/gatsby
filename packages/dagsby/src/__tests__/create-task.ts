import createTask from "../create-task"
import path from "path"

describe(`create task`, () => {
  it(`creates a simple task`, async () => {
    const task = await createTask({
      func: args => `hello ${args.name}!`,
      argsSchema: [
        {
          name: `name`,
          type: `string`,
        },
      ],
    })

    expect(typeof task.func).toBe(`string`)
    expect(typeof task.funcDigest).toBe(`number`)
    expect(typeof task.digest).toBe(`number`)

    console.log(task)
    expect(task).toMatchSnapshot()
  })

  it(`creates a simple task with files`, async () => {
    const task = await createTask({
      func: args => `hello ${args.name}!`,
      argsSchema: [
        {
          name: `name`,
          type: `string`,
        },
      ],
      files: {
        text: {
          originPath: path.join(__dirname, `mocks`, `hello.txt`),
        },
      },
      dependencies: {
        lodash: `latest`,
      },
    })

    console.log(task)
    expect(typeof task.files.text.digest).toBe(`string`)

    expect(task).toMatchSnapshot()
  })
})
