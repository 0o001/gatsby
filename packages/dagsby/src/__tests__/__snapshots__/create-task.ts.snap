// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`create task creates a simple task 1`] = `
Object {
  "argsSchema": Object {
    "fields": Array [
      Object {
        "name": "name",
        "type": "string",
      },
    ],
    "type": "record",
  },
  "digest": 72259,
  "func": "args => \`hello \${args.name}!\`",
  "funcDigest": 19923,
}
`;

exports[`create task creates a simple task with files 1`] = `
Object {
  "argsSchema": Object {
    "fields": Array [
      Object {
        "name": "name",
        "type": "string",
      },
    ],
    "type": "record",
  },
  "dependencies": Object {
    "lodash": "latest",
  },
  "digest": 95176,
  "files": Object {
    "text": Object {
      "digest": "6b8c327f0fc6f470c030a5b6c71154c5",
      "originPath": "<PROJECT_ROOT>/packages/dagsby/src/__tests__/mocks/hello.txt",
    },
  },
  "func": "args => \`hello \${args.name}!\`",
  "funcDigest": 19923,
}
`;
